hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -469,11 +469,38 @@ public abstract class AbstractEntityPersister
 	private List<UniqueKeyEntry> uniqueKeyEntries = null; //lazily initialized
 	private ConcurrentHashMap<String,SingleIdArrayLoadPlan> nonLazyPropertyLoadPlansByName;
 
+	/**
+	 * A factory for the creation of an EntityMetamodel.
+	 * <p>
+	 * Hibernate Reactive uses it to pass its own entity metamodel class
+	 * and adapt the identifier generators.
+	 */
+	public static class EntityMetamodelFactory {
+		public EntityMetamodel createEntityMetamodel(
+				PersistentClass persistentClass,
+				EntityPersister persister,
+				RuntimeModelCreationContext creationContext) {
+			return new EntityMetamodel( persistentClass, persister, creationContext );
+		}
+	}
```

Comment:
@DavideD instead of introducing this new class, could you just add a new `createEntityMetamodel()` operation to `RuntimeModelCreationContext`?

Or, if that doesn't work, could you at least make it a `@FunctionInterface`?

Reply:
I will check